src/components/TransactionsInterpreter/TransactionsInterpreter.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  ExtendedTransactionType,
+  TokenArgumentType,
+  UITransactionType
+} from './helpers/types';
+import getTransactionMethod from './helpers/getTransactionMethod';
+import { getTransactionTokens } from './helpers/getTransactionTokens';
+import urlBuilder from './helpers/urlBuilder';
+import { getDenominatedValue } from './helpers/getDenominatedValue';
+import { getNetworkLink } from './helpers/getNetworkLink';
+import { getEgldLabel, getTokenFromData, timeRemaining } from 'utils';
+import { isContract } from './helpers/isContract';
+import { getTransactionDirection } from './helpers/getTransactionDirection';
+import { NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS } from 'constants/transaction-interpreter';
+
+export type DenominationConfig = {
+  egldLabel?: string;
+  decimals?: number;
+  denomination?: number;
+  showLastNonZeroDecimal?: boolean;
+  showLabel?: boolean;
+  token?: string;
+};
+
+export function parseTransactions(
+  transactions: UITransactionType[],
+  address: string,
+  denominationConfig: DenominationConfig = {
+    egldLabel: 'EGLD'
+  },
+  numInitCharactersForScAddress: number = NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS
+): ExtendedTransactionType[] {
+  return transactions.map((transaction) => {
+    return processTransaction(
+      transaction,
+      address,
+      denominationConfig,
+      numInitCharactersForScAddress
+    );
+  });
+}
+
+export function processTransaction(
+  transaction: UITransactionType,
+  address: string,
+  denominationConfig: DenominationConfig = {
+    egldLabel: 'EGLD'
+  },
+  numInitCharactersForScAddress: number = NUMBER_OF_CHARACTERS_FOR_SMART_CONTRACT_ADDRESS
+): ExtendedTransactionType {
+  const tokenIdentifier =
+    transaction.tokenIdentifier ??
+    (getTokenFromData(transaction.data).tokenId || getEgldLabel());
```

Comment:
better use default than store.getState. This function will not be testable in isolation with `getEgldLabel()`